tasks/__init__.py

```diff
@@ -346,93 +346,93 @@ def sharejs(host=None, port=None, db_url=None, cors_allow_origin=None):
 
     share_server = os.path.join(settings.ADDON_PATH, 'wiki', 'shareServer.js')
-    run('node {0}'.format(share_server))
+    ctx.run('node {0}'.format(share_server))
 
 
 @task(aliases=['celery'])
-def celery_worker(level='debug', hostname=None, beat=False):
+def celery_worker(ctx, level='debug', hostname=None, beat=False):
     """Run the Celery process."""
     cmd = 'celery worker -A framework.celery_tasks -l {0}'.format(level)
     if hostname:
         cmd = cmd + ' --hostname={}'.format(hostname)
     # beat sets up a cron like scheduler, refer to website/settings
     if beat:
         cmd = cmd + ' --beat'
-    run(bin_prefix(cmd), pty=True)
+    ctx.run(bin_prefix(cmd), pty=True)
 
 
 @task(aliases=['beat'])
-def celery_beat(level='debug', schedule=None):
+def celery_beat(ctx, level='debug', schedule=None):
     """Run the Celery process."""
     # beat sets up a cron like scheduler, refer to website/settings
     cmd = 'celery beat -A framework.celery_tasks -l {0} --pidfile='.format(level)
     if schedule:
         cmd = cmd + ' --schedule={}'.format(schedule)
-    run(bin_prefix(cmd), pty=True)
+    ctx.run(bin_prefix(cmd), pty=True)
 
 
 @task
-def rabbitmq():
+def rabbitmq(ctx):
     """Start a local rabbitmq server.
 
     NOTE: this is for development only. The production environment should start
     the server as a daemon.
     """
-    run('rabbitmq-server', pty=True)
+    ctx.run('rabbitmq-server', pty=True)
 
 
 @task(aliases=['elastic'])
-def elasticsearch():
+def elasticsearch(ctx):
     """Start a local elasticsearch server
 
     NOTE: Requires that elasticsearch is installed. See README for instructions
     """
     import platform
     if platform.linux_distribution()[0] == 'Ubuntu':
-        run('sudo service elasticsearch start')
+        ctx.run('sudo service elasticsearch start')
     elif platform.system() == 'Darwin':  # Mac OSX
-        run('elasticsearch')
+        ctx.run('elasticsearch')
     else:
         print('Your system is not recognized, you will have to start elasticsearch manually')
 
 @task
-def migrate_search(delete=False, index=settings.ELASTIC_INDEX):
+def migrate_search(ctx, delete=False, index=settings.ELASTIC_INDEX):
     """Migrate the search-enabled models."""
     from website.search_migration.migrate import migrate
     migrate(delete, index=index)
 
 @task
-def rebuild_search():
+def rebuild_search(ctx):
     """Delete and recreate the index for elasticsearch"""
-    run('curl -s -XDELETE {uri}/{index}*'.format(uri=settings.ELASTIC_URI,
+    ctx.run('curl -s -XDELETE {uri}/{index}*'.format(uri=settings.ELASTIC_URI,
                                              index=settings.ELASTIC_INDEX))
-    run('curl -s -XPUT {uri}/{index}'.format(uri=settings.ELASTIC_URI,
+    ctx.run('curl -s -XPUT {uri}/{index}'.format(uri=settings.ELASTIC_URI,
                                           index=settings.ELASTIC_INDEX))
     migrate_search()
```

Comment:
You will need to pass `ctx` to this function and any other task that is called like this, since `ctx` is a required argument.